### PR TITLE
feat(contact-us): Contact Us header and background

### LINKED ISSUE #185: Contact page header and background
* [x] Styles and markup for the Contact Us page header.
* [x] Display background image for Contact Us.
* [x] Implement Make Contact button.
* [x] Should work on mobile and desktop.
~* [ ] Purchase design assets listed in the design below.~

image

Design: https://www.figma.com/file/YApYHX8kQ0WI0i7NcxnrY2vL/BaseTwo-Contact-Page

### PR DESCRIPTION
**Pull request by csteinert**
Items of note:

- I was too ambitious with Grid, previously. We don't need it, this branch removes it. It will be missed. ⚱️ 
  - Doing so caused some rework in the Hero section.
- `gradient` is now a styling option for Button.

Closes #185

### CODE REVIEW

File: src/components/atoms/BaseTwoLogo.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import styled, { css } from 'styled-components';
+
+const circleMixin = css`
+  background: black;
```
Comment (kalongthewires) [inline src/components/atoms/BaseTwoLogo.jsx]: Should this be #000 for consistency?

File: src/components/atoms/BaseTwoLogo.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  display: flex;
+  flex-direction: row;
+`;
+
+const BaseTwoLogo = () => (
+  <LogoWrapper>
+    <SolidCircle />
+    <StripedCircle />
+  </LogoWrapper>
+);
```
Comment (kalongthewires) [inline src/components/atoms/BaseTwoLogo.jsx]: Fancy.

File: src/components/atoms/Button.jsx [hunk 3]
```diff
@@ -1,26 +1,40 @@
...
-  background-color: ${themed('color.primary')};
-  border: 2px solid #80cbc4;
-  border-radius: 2px;
-  color: ${flow(themed('color.primary'), readableColor)};
+  ${themed('font.sansSerif')};
+  background-color: ${themed('color.transparent')};
+  border: 1px solid ${themed('color.link')};
+  color: ${themed('color.link')};
   cursor: pointer;
+  font-size: 2.625em;
```
Comment (kalongthewires) [inline src/components/atoms/Button.jsx]: Button is disproportionately large of mobile. Looks like text elsewhere is in rems.

File: src/components/atoms/Button.jsx [hunk 4]
```diff
@@ -38,7 +52,7 @@ const Button = styled.button`
 
   &:focus,
   &:hover {
-    background-color: ${flow(themed('color.primary'), lighten(0.1))};
+    background-color: ${themed('color.linkHover')};
```
Comment (kalongthewires) [inline src/components/atoms/Button.jsx]: Wondering if the color contrast between the background and the text is sufficient? Looks like the submit button in the form has a different hover effect as well -- are they meant to be different?

File: src/components/organisms/ContactHeader.jsx [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+        </LocationText>
+        <Earth src="img/planet-earth.jpg" alt="Planet Earth" />
+        <LocationTagline>
+          Thanks to satellites and the internet, we can work with you wherever
+          you are.
+        </LocationTagline>
+        <Button gradient onClick={onContact}>
+          Make Contact
+        </Button>
+        <Scroll src="img/scroll-arrows.svg" alt="" />
```
Comment (kalongthewires) [inline src/components/organisms/ContactHeader.jsx]: The scroll arrows under the satellite in the hero are small compared to the arrows here (particularly noticeable on mobile). Not sure if that's intentional.

Also, should this have alt text?

File: src/components/organisms/ContactHeader.jsx [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+        <LocationText>
+          is located in <strong>Columbus, OH</strong> and{' '}
+          <strong>Pittsburgh, PA</strong>.
+        </LocationText>
+        <Earth src="img/planet-earth.jpg" alt="Planet Earth" />
+        <LocationTagline>
+          Thanks to satellites and the internet, we can work with you wherever
+          you are.
+        </LocationTagline>
+        <Button gradient onClick={onContact}>
```
Comment (kalongthewires) [inline src/components/organisms/ContactHeader.jsx]: Should clicking the button do something? Or is this functionality being added separately?

File: src/components/organisms/ContactHeader.jsx [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+    margin: 0 auto;
+    max-width: 90%;
+    min-width: 10em;
+
+    ${mediaQuery.small`
+    max-width: 600px;
+  `};
+
+    ${mediaQuery.medium`
+    max-width: 615px;
```
Comment (kalongthewires) [inline src/components/organisms/ContactHeader.jsx]: Seeing a lot of different units for things. Wondering if there's some rule you're going by to determine whether to use px, vws, rem, or em?

File: src/components/organisms/Hero.jsx [hunk 8]
```diff
@@ -3,79 +3,51 @@ import styled from 'styled-components';
...
-  grid-row-end: 2;
+  height: 100vh;
+  padding: 3.75em 0 0 0;
+  position: relative;
+  text-align: center;
 
   ${mediaQuery.small`
     background-image: url('img/backgrounds/hero/b2-hero-background-960.png');
     background-size: 120%;
+    min-height: calc((100vw * ${BGAspectRatio}) * 1.2);
```
Comment (kalongthewires) [inline src/components/organisms/Hero.jsx]: What is the 1.2 value here? Magic number?

File: src/components/atoms/BaseTwoLogo.jsx [hunk 9]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
```
Comment (kalongthewires) [inline src/components/atoms/BaseTwoLogo.jsx]: Should B2LogoText be renamed to BaseTwoLogoText for consistency?

File: src/components/organisms/Hero.jsx (hunk 8 above)
Comment (csteinert) [reply src/components/organisms/Hero.jsx]: Aspect ratio on the background image.

File: src/components/atoms/Button.jsx (hunk 4 above)
Comment (csteinert) [reply src/components/atoms/Button.jsx]: Not sure, to my knowledge there wasn't a mock of a hovered button. I set the submit hover styles to be the default, so now all buttons have them. 👍

File: src/components/organisms/ContactHeader.jsx (hunk 7 above)
Comment (csteinert) [reply src/components/organisms/ContactHeader.jsx]: The PX were lazy, I'm replacing them now with the `em` function from Polished.

My thought process around em, vws, or % is:
- Use EMs when I know the exact size I want an element to be/not to exceed (translate from px if necessary) or when sizing typography relative to the 16px base size.
- Use VW/VH for high-level layout elements where viewport size is a design factor (eg our background images).
- Use percentages to resize elements relative to the parent size.

And now that I've written that, I'll make sure I abide by it! 😂

Comment (csteinert) [reply src/components/organisms/ContactHeader.jsx]: I think the REMs were coming from Mandy's code. I never know which way to go with those. Gonna stick with EMs for now, can discuss when she's back.

File: src/components/organisms/ContactHeader.jsx (hunk 6 above)
Comment (csteinert) [reply src/components/organisms/ContactHeader.jsx]: It should navigate to the contact form below, probably via a link to an ID (which won't exist until the bottom section is done). But now I'm having an existential crisis where I don't think this should be a button element. It's a link. Might have to make a new element of "link button" or something to make this work properly.

Can I punt this to #197?

Comment (kalongthewires) [reply src/components/organisms/ContactHeader.jsx]: Agree and yes.

File: src/components/organisms/ContactHeader.jsx (hunk 7 above)
Comment (kalongthewires) [reply src/components/organisms/ContactHeader.jsx]: 😆 Sounds good to me.

**Pull request merged**